source/common/network/listen_socket_impl.cc

```diff
@@ -19,9 +19,9 @@ void ListenSocketImpl::doBind() {
 }
 
 // TODO(wattli): remove this once the admin port is updated with address.
-TcpListenSocket::TcpListenSocket(const std::string& address, bool bind_to_port) {
+TcpListenSocket::TcpListenSocket(const Address::InstancePtr& address, bool bind_to_port) {
```

Comment:
Just pass Address::InstancePtr here (passing by const& can lead to subtle bugs). Same below.